### PR TITLE
Remove `BBF_TRY_BEG`

### PR DESCRIPTION
**Pull request by SingleAccretion**
The flag duplicates information already present in the EH table.

Also delete some (not all) remnants of verification from `impVerifyEHBlock`.

### CODE REVIEW

**Comment by SingleAccretion** [comment]
Failures look like #93404.

[No diffs](https://dev.azure.com/dnceng-public/public/_build/results?buildId=436331&view=ms.vss-build-web.run-extensions-tab), but a minor TP win, @dotnet/jit-contrib .

File: src/coreclr/jit/jiteh.cpp [hunk 1]
```diff
@@ -3499,12 +3498,6 @@ void Compiler::fgVerifyHandlerTab()
 #endif // FEATURE_EH_FUNCLETS
         }
 
-        // Only the first block of 'try' regions should have BBF_TRY_BEG set.
-        if (!blockTryBegSet[block->bbNum])
```
Comment (BruceForstall) [inline src/coreclr/jit/jiteh.cpp:3503]: The only use of `blockTryBegSet` currently is in this check. You could remove it entirely.

File: src/coreclr/jit/fgopt.cpp [hunk 2]
```diff
@@ -6400,10 +6400,9 @@ bool Compiler::fgUpdateFlowGraph(bool doTailDuplication, bool isPhase)
                 goto REPEAT;
             }
 
-            /* Remove unreachable or empty blocks - do not consider blocks marked BBF_DONT_REMOVE or genReturnBB block
-             * These include first and last block of a TRY, exception handlers and RANGE_CHECK_FAIL THROW blocks */
-
-            if ((block->bbFlags & BBF_DONT_REMOVE) == BBF_DONT_REMOVE || block == genReturnBB)
```
Comment (BruceForstall) [inline src/coreclr/jit/fgopt.cpp:6406]: Is the `genReturnBB` check just unnecessary? (obsolete?)

**Review by BruceForstall — COMMENTED** [review]
That's a nice cleanup. A couple notes.

As a follow-up: do you have any thoughts on https://github.com/dotnet/runtime/issues/82336?

File: src/coreclr/jit/fgopt.cpp (hunk 2 above)
Comment (SingleAccretion) [reply src/coreclr/jit/fgopt.cpp:6406]: Unnecessary; the block is [already always marked](https://github.com/dotnet/runtime/blob/946c5245aea149d9ece53fd0debc18328c29083b/src/coreclr/jit/flowgraph.cpp#L2357-L2359) with `BBF_DONT_REMOVE`.

**Comment by SingleAccretion** [comment]
> As a follow-up: do you have any thoughts on #82336?

Great question :). I am looking at the EH-related flowgraph things and in particular the ability to remove dead protected regions and handlers.

File: src/coreclr/jit/jiteh.cpp (hunk 1 above)
Comment (SingleAccretion) [reply src/coreclr/jit/jiteh.cpp:3503]: Deleted.

**Comment by BruceForstall** [comment]
@SingleAccretion Can you fix the merge conflict?

fwiw, all the test failures are "known"

**Comment by SingleAccretion** [comment]
@BruceForstall conflict fixed.

**Review by BruceForstall — APPROVED** [review]

**Pull request merged**